### PR TITLE
Use en dash (–) for ranges in compat tables

### PR DESCRIPTION
**Pull request by foolip**
The en dash is narrower and more commonly used for ranges.

### CODE REVIEW

**Comment by foolip** [comment]
@peterbe can you review?

File: client/src/document/ingredients/browser-compatibility-table/feature-row.tsx
Comment (foolip) [inline client/src/document/ingredients/browser-compatibility-table/feature-row.tsx:144]: While in the area I wonder if I can sell a few more changes? It seems to me that using zero width spaces around the dash or possibly one of the narrow spaces mentioned in https://en.wikipedia.org/wiki/Whitespace_character#Hair_spaces_around_dashes would be better than the full space we now have.

If I could suggest anything, it would be `{labelFromString(added)}&hairsp;&ndash;&hairsp;{labelFromString(removed)}`

**Comment by peterbe** [comment]
@foolip You seem quite invested and clearly know a lot about this. And you've proven yourself to really care about BCD. 
Unless @ddbeck or @Elchi3 have anything to object to, I'd be happy to blindly take your advice without fully understanding it. 

PS. Today was the first time, after 20+ of web dev, heard of "thin space" and "hair space".

**Review by peterbe — APPROVED** [review]
Unless Daniel or Florian object, I'd be perfectly happy to let you merge whatever you think is best here. 

I do wonder, you PR does not use hair spaces but you're making a good argument for that. Do you want to go all the way and implement that too as part of this PR? Or do you want to do that as a follow-up once this is merged?
I'm basically a bit confused as to what to do with your "sell a few more changes" comment there.

**Comment by foolip** [comment]
Let me make the changes I'd like to see, to show it clearly. I'm not sure how to preview it, so some help with that would be great.

**Comment by foolip** [comment]
OK, I've made the change I think will work now. How do I tell if I broke the Internet?

**Comment by foolip** [comment]
> PS. Today was the first time, after 20+ of web dev, heard of "thin space" and "hair space".

Actually I didn't know there were two different thin spaces either, I found out as I looked for a thin space that's still a line breaking opportunity.

**Comment by peterbe** [comment]
> OK, I've made the change I think will work now. How do I tell if I broke the Internet?

If you're referring to CI, I had to click the "Approve run" since you're still a first-time contributor. But hopefully not for long!

If you're referring to actually testing it, you'll need to use `yari` locally and do `yarn dev` and test on localhost:3000.

**Comment by peterbe** [comment]
@foolip Does the CI error make sense? Do you know what you need to do with `prettier`?

**Comment by foolip** [comment]
A lint error I can figure out, and now I know how to preview locally too. I assumed I'd need the content repo to preview somehow.

**Comment by foolip** [comment]
I previewed this locally and I'm happy with it, a clear improvement IMHO. @peterbe over to you.

File: client/src/document/ingredients/browser-compatibility-table/feature-row.tsx
Comment (peterbe) [inline client/src/document/ingredients/browser-compatibility-table/feature-row.tsx:139]: So much simpler!

**Review by ddbeck — APPROVED** [review]
Works for me. Approving. Not sure what (if any) expectations there are around merging to yari, so I'll leave the actual merge up to y'all. Thank you!

**Comment by peterbe** [comment]
Excellent work @foolip !
I know it's you but perhaps you can look into setting up git signing of your commits.

**Comment by foolip** [comment]
> I know it's you but perhaps you can look into setting up git signing of your commits.

I use the GitHub web UI a lot for fixing things up. Is that really possible without just giving GitHub my private GPG key?

Also my assumption is that PRs will be squash merged (https://github.com/mdn/yari/commit/746472f145bcf6886b1323fa668a66625441c096 was) and then signature wouldn't carry along to the main branch.

**Comment by peterbe** [comment]
> I use the GitHub web UI a lot for fixing things up. Is that really possible without just giving GitHub my private GPG key?

I actually don't know. Never explored that. 



> Also my assumption is that PRs will be squash merged ([746472f](https://github.com/mdn/yari/commit/746472f145bcf6886b1323fa668a66625441c096) was) and then signature wouldn't carry along to the main branch.

That's right. We use (and recommend) that people sign their commits from a security perspective so you can know with confidence that it's YOU who made the PR. Anybody can make a commit under any name and email, in git, so someone could pretend to be you in a PR and because you're trusted, your PR doesn't get carefully scrutinized. 

What I've never understood about that is; how can someone make a GitHub PR on github.com under a different name?!? Doesn't the GitHub UI make sure that your username is that with which you verified your account etc. 
Cuz if a hacker created an account with username `fooolip` and stole your avatar and full name, they could just sign their own commits and we'd be equally tricked. 

All that to say, as long as a merger recognizes the username in the PR, I don't know how that's not secure/safe.

**Comment by foolip** [comment]
Indeed, you already know that https://github.com/foolip created the PR. And unless we've been to a key signing party, why should you trust that 0xF75964F29DC6C210 is more the Philip you know that my GitHub account?

**Comment by peterbe** [comment]
Still baffles me! 

If someone typo-squats your GitHub username (and copies your avatar) they might try and succeed to sneak a PR and I don't see how signing your git commits would help there. Because, as you say, I don't know more about your key than I do about someone else, who pretends to be you with a tiny difference in username.

Thankfully, now that GitHub can clearly say if someone is a first-time contributor, it stands out more. So if I see a friendly-looking PR from someone who looks like you (but small difference in username) makes a PR it would say "First-time contributor" again(!) and that should hopefully send me a red flag. But it's hard to pay that much careful attention. 

Also, I trust the "Files" tab in the GitHub PR UI. I scroll through all the changes and see that there's nothing "fishy". But it's equally easy for a hacker to subtle sneak in a rogue edit to the `yarn.lock` file because those are often hard (actually, almost impossible) to review.

**Comment by Ryuno-Ki** [comment]
> Also, I trust the "Files" tab in the GitHub PR UI. I scroll through all the changes and see that there's nothing "fishy". But it's equally easy for a hacker to subtle sneak in a rogue edit to the `yarn.lock` file because those are often hard (actually, almost impossible) to review.

That's my main safe guard. I even would comb through it if the PR author is called peterbe :-)
Because everyone is prone to make mistakes. That's where reviews shine.

I'm not sure, how much „Web of trust” is supported by GitHub. I believe, not that much.
You could defer to linked Twitter accounts or homepage from one's profile to increase plausability, but that's also fakeable.

**Pull request merged**
